File: binarySearch/java/BinarySearch.java
package binarySearch.java;
public class BinarySearch {

    public static int search(int key, int a[]) {
        return search(key, a, 0, a.length - 1);
    }

    private static int search(int key, int a[], int lo, int hi) {
        if (lo > hi)
            return -1;
            
        int mid = lo + (hi - lo) / 2;

        if (key < a[mid])
            return search(key, a, lo, mid - 1);
        else if (key > a[mid])
            return search(key, a, mid + 1, hi);
        else
            return mid;

    }

}

File: binarySearch/python/binarySearch.py
def binary_search(vector, target, lo=None, hi=None):

    if lo == None:
        lo = 0
        hi = len(vector) - 1

    if lo > hi:
        return -1
    
    mid = int(lo + (hi - lo) / 2)

    if target < vector[mid]:
        return binary_search(vector, target, lo, mid -1)
    elif target > vector[mid]:
        return binary_search(vector, target, mid + 1, hi)
    else:
        return mid


vector = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

print(binary_search(vector, 3))